pkg/predictor_source/inferenceservice_registry.go

```diff
@@ -94,24 +141,22 @@ func (isvcr InferenceServiceRegistry) Get(ctx context.Context, nname types.Names
 		p.Status.ActiveModelState = v1alpha1.Pending
 	}
 
-	_, frameworkSpec := inferenceService.Spec.Predictor.GetPredictorFramework()
-	storageUri := *frameworkSpec.StorageURI
 	p.Spec.Storage = &v1alpha1.Storage{}
-	if strings.HasPrefix(storageUri, "s3://") {
-		s3Uri := strings.TrimPrefix(storageUri, "s3://")
-		urlParts := strings.Split(s3Uri, "/")
-		bucket := urlParts[0]
-		modelPath := strings.Join(urlParts[1:], "/")
-		secretKey := inferenceService.ObjectMeta.Annotations[v1beta1.SecretKeyAnnotation]
-		if schemaPath, ok := inferenceService.ObjectMeta.Annotations[v1beta1.SchemaPathAnnotation]; ok {
-			p.Spec.SchemaPath = &schemaPath
-		}
-		p.Spec.Storage.S3 = &v1alpha1.S3StorageSource{
-			SecretKey: secretKey,
-			Bucket:    &bucket,
-		}
-		p.Spec.Path = modelPath
+	secretKey, bucket, modelPath, schemaPath, err := processInferenceServiceStorage(inferenceService, nname)
```

Comment:
Thanks @Tomcli looks great now... one more small thing I noticed - might be best for `schemaPath` and `bucket` to be pointers since they are in both CRDs and it would make sense to transfer `nil` values from IS to the corresponding Predictor fields (rather than converting `nil` to a pointer to empty string as is the case now).